src/Libraries/Microsoft.Extensions.AI/Functions/AIFunctionFactory.cs

```diff
@@ -471,6 +461,20 @@ private static Type GetReturnMarshaler(MethodInfo method, out Func<object?, Valu
 #endif
         }
 
+        private static readonly MethodInfo _taskGetResult = typeof(Task<>).GetProperty(nameof(Task<int>.Result), BindingFlags.Instance | BindingFlags.Public)!.GetMethod!;
```

Comment:
Not sure if using GetProperty(nameof()).GetMethod is preferable to doing GetMethod("get_Result") directly from a linker perspective.